webapp/src/components/assetEditor/editor.tsx

```diff
@@ -101,6 +101,14 @@ export class AssetEditor extends Editor {
                 });
                 break;
             case pxt.AssetType.Tilemap:
+                // for tilemaps, fill in all project tiles
+                const allTiles = pxt.react.getTilemapProject().getProjectTiles(asset.data.tileset.tileWidth, true);
```

Comment:
move "fill code" out of individual "create"/"edit" handlers; always fill in all available tiles before opening the field view